packages/frontend/src/contexts/game/reducer/index.js

```diff
@@ -23,6 +23,19 @@ export const gameReducer = (state, action) => {
             ? { ...tile, faceUp: !tile.faceUp }
             : tile
         ),
+        flipped: state.flipped.concat(action.payload.tile),
```

Comment:
We could probably also set the `isMatched` value if the `flipped` array has a length of 2.

The `produce` function from Immer could be used if it makes the syntax more readable.

```javascript
produce(state, draft => {
  if(state.flipped.length === 2){
    draft.tiles = draft.tiles.map(tile => state.flipped.contains(tile) ? ({ ...tile, isMatched: true }) : tile) 
  }
})
```

I'm not 100% on that syntax because I'm writing off the top of my head. I think adding the `companion` property with the index of the companion tile or maybe just the companion tile object itself would make it easier to check/switched the `isMatched` value.